hops/featurestore_impl/util/fs_utils.py

```diff
@@ -653,26 +653,28 @@ def _validate_metadata(name, dtypes, description):
         :name: the name of the feature group/training dataset
         :dtypes: the dtypes in the provided spark dataframe
         :description: the description
+        :featurestore_regex: Regex string to match featuregroup/training dataset and feature names with
 
     Returns:
         None
 
     Raises:
         :ValueError: if the metadata does not match the required validation rules
     """
-    name_pattern = re.compile("^[a-zA-Z0-9_]+$")
+    name_pattern = re.compile(featurestore_settings.featurestore_regex)
```

Comment:
You don't need to compile the regex every time you call the method.

Reply:
Moved compilation to settings